code_search/code-search.ipynb

```diff
@@ -0,0 +1,451 @@
+   "source": [
+    "## Export Tensorflow Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "ks apply code-search -c t2t-code-search-exporter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compute Function Embeddings\n",
+    "\n",
+    "In this step, we will use the exported model above to compute function embeddings via another `Dataflow` pipeline. A `Python 2` module `code_search.dataflow.cli.create_function_embeddings` has been provided for this purpose. A list of all possible arguments can be seen below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "python2 -m code_search.dataflow.cli.create_function_embeddings -h"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configuration\n",
+    "\n",
+    "First, select a Exported Model version from the `${WORKING_DIR}/output/export/Servo`. This should be name of a folder with UNIX Seconds Timestamp like `1533685294`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "MODEL_VERSION = '1533685294'\n",
+    "\n",
+    "# DO NOT MODIFY. The following statements are needed to support\n",
+    "# environment variables in a bash cell.\n",
+    "%env MODEL_VERSION $MODEL_VERSION"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run the Dataflow Job for Function Embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "python2 -m code_search.dataflow.cli.create_function_embeddings \\\n",
+    "        --runner DataflowRunner\n",
+    "        --project \"${PROJECT}\" \\\n",
+    "        --target_dataset \"${TARGET_DATASET}\" \\\n",
+    "        --problem github_function_docstring \\\n",
+    "        --data_dir \"${WORKING_DIR}/data\" \\\n",
+    "        --saved_model_dir \"${WORKING_DIR}/output/export/Servo/${MODEL_VERSION}\" \\\n",
+    "        --job_name compute-function-embeddings\n",
+    "        --temp_location \"${WORKING_DIR}/data/dataflow/temp\" \\\n",
+    "        --staging_location \"${WORKING_DIR}/data/dataflow/staging\" \\\n",
+    "        --worker_machine_type \"${WORKER_MACHINE_TYPE}\" \\\n",
+    "        --num_workers \"${NUM_WORKERS}\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When completed successfully, this should create another table in the same `BigQuery` dataset which contains the function embeddings for each existing data sample available from the previous Dataflow Job. Additionally, it also dumps a CSV file containing metadata for each of the function and its embeddings."
```

Comment:
Can we add a cell to query bigquery and display the first 10 rows? Ideally we'd leverage BQ integration so that this shows up as a nice widget.

Omar and Bradley will know.

Reply:
This would be a good addition. I'm following the email thread. Thanks.